File: Stacks/stack.py
class Stack:
    def __init__(self,MAX):
        self.stack=[]
        self.top=-1
        self.size=MAX

    def isEmpty(self):
        return self.top == -1

    def isFull(self):
        return self.top == self.size-1
    
    def push(self,element):
        if(not self.isFull()):
            self.top=self.top+1
            self.stack.append(element)
            print("Item Added is ",element)
        else:
            print("Stack is Full")    

    def pop(self):
        if(not self.isEmpty()):
            print("Item Removed is ",self.stack.pop())
            self.top=self.top-1
        else:
            print("Stack is Empty")      

    def printStack(self):
        for i in range(self.top+1):
            print(self.stack[i]," ")


stack=Stack(5)
stack.push(10)
stack.push(20)
stack.push(30)
stack.printStack()
stack.push(40)
stack.push(50)
stack.pop()
stack.printStack()

File: Stacks/stack.cpp
#include<iostream>
using namespace std;

#define MAX 5

class Stack{
    private:
       int top;
    public:
       int arr[MAX];
       Stack(){
        top=-1;
       }
       bool isFull(){
           return top==MAX-1;
       }
       bool isEmpty(){
           return top==-1;
       }
       void push(int element){
            if(!isFull()){
                top+=1;
                arr[top]=element;
                cout<<arr[top]<<" is added successfully\n";
            }     
            else{
                cout<<"Stack is Full";
            }
       }  
       void pop(){
           if(!isEmpty()){
                int element=arr[top];
                top-=1;
                cout<<element<<" is removed successfully\n";
            }
            else{
                cout<<"Stack is Empty";
            }
       }
       void printStack(){
           if(!isEmpty()){
               for(int i=0;i<=top;i++){
                   cout<<arr[i]<<" ";
               }
           }
       }
};

int main(){
  Stack stack;
  stack.push(10);
  stack.push(20);
  stack.push(30);
  stack.push(40);
  stack.pop();
  stack.printStack();
  return 0;
}